### PR TITLE
Adding support for Group Emails

### PR DESCRIPTION
**Pull request by daniel-samson**
## Description

This change now shows the folders for group emails. There are issues with the user profile settings regarding folder management. The user can easily tell which folders are for which mail boxes. Perhaps we need a tree view with checkboxes instead.

There may still be issues in the folder management. Which directly effects what is displayed on the folders modal popup.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: modules/Emails/javascript/EmailUI.js [hunk 1]
```diff
@@ -891,7 +891,7 @@ SE.accounts = {
 
             AjaxObject._reset();
             AjaxObject.target = 'frameFlex';
-            AjaxObject.startRequest(callbackAccount, urlStandard + '&emailUIAction=saveIeAccount');
+            AjaxObject.startRequest(callbackAccount, urlStandard + '&emailUIAction=saveIeAccount&XDEBUG_SESSION_START=13537');
```
Comment (samus-aran) [inline modules/Emails/javascript/EmailUI.js]: Remove Debugging

File: modules/Emails/javascript/EmailUI.js [hunk 2]
```diff
@@ -2039,19 +2039,21 @@ SE.folders = {
...
-            	AjaxObject.startRequest(AjaxObject.accounts.callbackCheckMailProgress, urlStandard +
-                                '&emailUIAction=checkEmailProgress&ieId=' + SE.accounts.ieIds[0] + "&currentCount=0");
-            } else {
-               SUGAR.hideMessageBox();
-            }
-        } else {
-            // wait 5 secs before trying again.
-            SE.folders.checkingMail = setTimeout("SE.folders.startEmailAccountCheck();", 5000);
-        }
+       // if(!AjaxObject.requestInProgress()) {
```
Comment (samus-aran) [inline modules/Emails/javascript/EmailUI.js]: Remove remove remove

File: modules/InboundEmail/language/en_us.lang.php [hunk 3]
```diff
@@ -199,5 +199,18 @@
     'LBL_WARNING_CHANGING_AUTO_IMPORT' => 'Warning: You are modifying your automatic import setting which may result in loss of data.',
     'LBL_WARNING_CHANGING_AUTO_IMPORT_WITH_CREATE_CASE' => 'Warning: Auto import must be enabled when automatically creating cases.',
 	'LBL_EDIT_LAYOUT' => 'Edit Layout' /*for 508 compliance fix*/,
+    'LBL_LIST_TITLE_MY_DRAFTS'	=> 'Drafts',
```
Comment (samus-aran) [inline modules/InboundEmail/language/en_us.lang.php]: Perhaps in the last commit you make we should format the files.

File: include/SugarFolders/SugarFolders.php [hunk 4]
```diff
@@ -1120,6 +1125,7 @@ public function retrieve($id)
             }
 
             return true;
+            $new_with_id  = false;
```
Comment (daniel-samson) [inline include/SugarFolders/SugarFolders.php]: I meant to put it above the return statement.

File: modules/Emails/EmailUIAjax.php [hunk 5]
```diff
@@ -996,12 +996,29 @@
         break;
 
     case "setFolderViewSelection": // flows into next case statement
+        global $db;
         $GLOBALS['log']->debug("********** EMAIL 2.0 - Asynchronous - at: setFolderViewSelection");
         $viewFolders = $_REQUEST['ieIdShow'];
+        $userSubscriptions = $email->et->folder->getSubscriptions($focus);
```
Comment (daniel-samson) [inline modules/Emails/EmailUIAjax.php]: Admins should be able to set which folders for a user. In order to achieve this we need to pass the user that is being edited. I have added the lines required.

File: modules/Emails/EmailUIAjax.php [hunk 6]
```diff
@@ -996,12 +996,29 @@
...
         $current_user->setPreference('showFolders', base64_encode(serialize($viewFolders)), '', 'Emails');
         $tree = $email->et->getMailboxNodes(false);
         $return = $tree->generateNodesRaw();
         $out = $json->encode($return);
+
+//        $email->et->folder->clearSubscriptions();
```
Comment (daniel-samson) [inline modules/Emails/EmailUIAjax.php]: Fixed

**Comment by horus68** [comment]
@daniel-samson the file `modules/InboundEmail/language/en_us.lang.php` has several double spaces in the middle of a sentence. Can you please remove those as it causes issues when searching for text on translations.

**Comment by gymad** [comment]
@horus68 a hint: how do you do the search? If you have a regex search in it, you can use the \s+ to find any white spaces in the text which is including the double spaces too (I hope it's help till we find out to solve this)

**Comment by daniel-samson** [comment]
@horus68 Can you please explain to me what the issue is?

As I don't understand how double spacing can effect translations. Especially when we have 4 spaces for the tabs. Wouldn't that also be a problem?

We have never had to care about this before. Why is this an issue now?

**Comment by daniel-samson** [comment]
@gymad @samus-aran please see #3471 for amendments to this pull request.

**Comment by horus68** [comment]
@daniel-samson I was referring double spaces inside a value (sentence to be translated), not in tabs for formating.
When searching Crowdin translation platform for the sentence:
**Testing SSL may take a long time. Please be patient** the string will not be found as actually the english text has a double space after the "."
**Testing SSL may take a long time.  Please be patient**
Also some of those double spaces (not all!) shows in crowdin with rightwards arrow to bar sign `⇥` . Maybe due to been edited in Mac.
See my issue on other files (and PR) https://github.com/salesagility/SuiteCRM/issues/3427

**Comment by daniel-samson** [comment]
@horus68 The first problem is that the double spacing is a cultural thing in the English language. It comes from when we had the printing press. So for many writers of English this feels very natural to add a double space. In some legal documents this denotes meaning.  We realise that this is a matter of opinion that has changed which weren't aware of. So that's something we have learned

The second issue is changing / updating the language files. As you know, there are quite a few language files. If we are to update the languages files in this way, we should do them all at once and then make a pull request.

For my pull requests  I will not be changing the double spaces for the above reasons. I will not put double spaces the values. If you find something in my changes that do have double spaces then I will then change it.

Going forward I think we could create a PHP script to remove all the double space issues in values section of SuiteCRM. Then we could make one might pull request to handle deal with these issues.

**Comment by horus68** [comment]
@daniel-samson I have already created the PR here, but not merged yet:
https://github.com/salesagility/SuiteCRM/pull/3429

**Comment by daniel-samson** [comment]
@horus68 Great! Does this contain all of the files with double spaces?

**Comment by horus68** [comment]
@daniel-samson yes for language strings

on the license header there is also an issue https://github.com/salesagility/SuiteCRM/issues/3434
I did,'t created a PR as license header for language strings is already in the make on a different combined license language change PR #3348

**Comment by daniel-samson** [comment]
Thanks @horus68 ! :D

**Pull request merged**